### DISCUSSION TITLE
VIO device not found

### DISCUSSION THREAD

**Discussion by jgalinat** [thehighnibble/thehighnibble.github.io#80] [S-132 add-on for IMSAI & Cromemco replicas]
I am normally using the TTY1 device as the console.   I wanted to check out running the VIO device as the console so I issued the CP/M command STAT con:=crt: and I got the response “NO VIO”.  I also saw that there was a program on the A: disk called VMODE.COM which I assumed was to set various video modes. When I run VMODE.COM  I get the response “NO VIO FOUND”.   If I hit the Ctrl+Alt+F4 combination nothing happens. All the other Ctrl+Alt function keys work as expected. 

I followed the configuration instructions in the for the S-132 and edited the boot.cnf to include to make the S-132 the first preference for the TTY: & VIO: devices and also simultaneously use the Web Desktop devices if connected, use:

### SIO-2 default port mappings (HAL)
SIO1.portA.device=S132TTY+,WEBTTY,UART0
SIO1.portB.device=S132VIO+,VIOKBD

Any suggestions on what I am missing?

![AEC73968-BAB0-48C3-B779-65E5B282B525](https://github.com/thehighnibble/thehighnibble.github.io/assets/131741410/b5f9ac8e-ce11-468f-9f97-f598feb608a1)

**Comment by thisspaceleftblank** [comment]
You have to be booting with a ROM that supports VIO.. Change what ROM you boot with.

**Reply by jgalinat** [reply]
Thank you for pointing me in the right direction. I was using the default memory map (1) so I changed it to memory map 2 using the NVS startup configuration and now it works.

**Comment by guibrush1** [comment]
Hi, I'm facing the same issue, and I muss say that I didn't manage to understand how to change this memory map, I don't find something like that in the config file. Can you point me to the right direction? Thanks

**Reply by thisspaceleftblank** [reply]
Turn the power off to your device, hold the EXAMINE toggle up and hit a reset (there is one on the side and one in the back, pick one).

The display should come up with the 4 lights on the right scrolling from right to left and the 16 bits of address on the bottom will display your current config.

Mine show bits 11, 8, 7, and 3 on.  Looking at the chart on the website under "Configuration" my config is 3 - Station Mode, 7 - Unlimited Speed, 8 - Memory Map 1 (from your system.conf on the emulator), and 11 - NVS_BANK_ROM enabled.

My configuration allows it to connect to my wifi router and deliver up Memory Map 1 which will boot the RDOS ROM... from there a 'ba' command in the terminal will boot disk in drive a.

Anyway, from the configuration setup on the front panel you toggle up all the switches that you want ON and toggle DEPOSIT up to store the settings. So to match my configuration, 3, 7,8 and 11 are up, all others are down, toggle 'DEPOSIT' up and its stored.. hit reset.

I am currently set up for the Cromemco and that is the chart I'm looking at for the values. The IMSAI may be different but the process is the same.

**Reply by guibrush1** [reply]
thanks! This did the trick. I thought it's a parameter to change in the config file, I was totally wrong.
Now I have the SIO working :-)
